proxy/src/fully_qualified_authority.rs

```diff
@@ -52,19 +56,34 @@ impl FullyQualifiedAuthority {
         };
 
         // Rewrite "$name.$namespace" -> "$name.$namespace.svc".
-        let (has_svc, append_svc) = if let Some(part) = parts.next() {
-            (part.eq_ignore_ascii_case("svc"), false)
+        let append_svc = if let Some(part) = parts.next() {
+            if !part.eq_ignore_ascii_case("svc") {
+                // if not "$name.$namespace.svc", treat as external
+                return None;
+            } else {
+                false
+            }
+        } else if has_explicit_namespace {
+            true
+        } else if namespace_to_append.is_none() {
+            // if not "*.svc" and no default namespace, treat as external
```

Comment:
To be precise the comment would have to be `If not "*.svc" and there's no namespace, treat as external.` However, that's just duplicating the logic in the `if {} else {}`. I suggest the comment be `// We can't append ".svc" without a namespace, so treat as external.`